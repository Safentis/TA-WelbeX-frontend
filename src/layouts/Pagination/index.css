.pagination {
  display: flex;
  justify-content: center;
}

.pagination__item {
  margin: 0 0.5rem;
}
