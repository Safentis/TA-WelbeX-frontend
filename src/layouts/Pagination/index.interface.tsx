import { Entry } from '../../application/App/App.interface';

export interface Props {
  entries: Entry[];
  entriesOnPage: number;
  currentPage: number;
  handlePagination: (num: number) => void;
}
