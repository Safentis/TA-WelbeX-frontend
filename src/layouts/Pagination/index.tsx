import React, { FC } from 'react';

import Button from '../../components/Button/Button';

import './index.css';
import { Props } from './index.interface';
import { Entry } from '../../application/App/App.interface';

/**
 * Pagination
 * @param {Entry[]} entries
 * @param {number} entriesOnPage
 * @param {number} currentPage
 * @param {(num: number) => void} handlePagination
 * @returns {React.ReactElement} returns react element
 */

// prettier-ignore
const Pagination: FC<Props> = ({ entries, entriesOnPage, currentPage, handlePagination }): React.ReactElement => {
  //* -------------------------------------------------------------------------
  //* All page numbers
  const activeBtnCls: string = 'button_active';
  const pageNumbers: number[] = [];
  const division: number = Math.ceil(entries.length / entriesOnPage);

  //* In this case we get all page numbers  
  for (let num = 1; num <= division; num++) {
    pageNumbers.push(num);
  }

  return (
    <ul className="pagination">
      {pageNumbers.map((num: number) => (
        <li className="pagination__item">
          <Button 
            className={currentPage === num ? activeBtnCls : ''} 
            key={num} 
            handleClick={() => handlePagination(num)}>
            {num}
          </Button>
        </li>
      ))}
    </ul>
  );
};

export default Pagination;
