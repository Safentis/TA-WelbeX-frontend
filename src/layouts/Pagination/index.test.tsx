import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import Pagination from './index';

import { Props } from './index.interface';

const setUp = (props: Props) => {
  return shallow(<Pagination {...props} />);
};

describe('<Pagination />', () => {
  let component: ShallowWrapper<
    any,
    Readonly<{}>,
    React.Component<{}, {}, any>
  >;
  let props: Props = {
    entries: [],
    entriesOnPage: 1,
    currentPage: 1,
    handlePagination: () => {},
  };

  beforeEach(() => {
    component = setUp(props);
  });

  it('component does match to the snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  describe('Markup: ', () => {
    it('has 1 ul element', () => {
      expect(component.find('ul')).toBeTruthy();
    });

    it('has 1 li element', () => {
      expect(component.find('li')).toBeTruthy();
    });

    it('has 1 Button element', () => {
      expect(component.find('Button')).toBeTruthy();
    });
  });
});
