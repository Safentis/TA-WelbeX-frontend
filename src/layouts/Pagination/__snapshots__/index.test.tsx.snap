// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> component does match to the snapshot 1`] = `
<ul
  className="pagination"
/>
`;
